src/toil/batchSystems/slurm.py

```diff
@@ -444,6 +442,37 @@ def prepareSbatch(self,
             if cpu is not None:
                 sbatch_line.append(f'--cpus-per-task={math.ceil(cpu)}')
 
+            if gpus:
+                sbatch_line = sbatch_line[:1] + [f'--gres=gpu:{gpus}'] + sbatch_line[1:]
+
+                lowest_priority_gpus = sorted([i for i in self.boss.partitions if i[1]], key=lambda x: x[4])
+                new_partition_name = lowest_priority_gpus[0][0]  # get name of lowest priority partition that supports gpus
+                if not any(line.startswith("--partition") for line in sbatch_line):
```

Comment:
I think `sbatch_line` is one (command) "line", so the elements are probably `option` and not `line`.
```suggestion
                if not any(option.startswith("--partition") for option in sbatch_line):
```